### PR TITLE
Mark radialmenu

### PR DESCRIPTION
**Pull request by mroukema**
Enhancements to arc drawer. Can draw sections of an arc with a separate rim component. rim width and total radius are variable.

Rim and body or arc can have separate textures.

### CODE REVIEW

File: Virtual Keyboard/Assets/ArcMeshDrawer.cs [hunk 1]
```diff
@@ -5,172 +5,156 @@ public class ArcMeshDrawer : MonoBehaviour {
...
-		normals = new Vector3[size + 1];
-		uv = new Vector2[size + 1];
-		
-		for (int i = 0; i < uv.Length; i++)
-			uv[i] = new Vector2(0, 0);
-		for (int i = 0; i < normals.Length; i++)
-			normals[i] = new Vector3(0, 1, 0);
+	public void createMeshes() {
+		calculateSizes ();
+		computeSharedVeticies ();
```
Comment (eh-dub) [inline Virtual Keyboard/Assets/ArcMeshDrawer.cs]: computeSharedVeticies should be spelled computeSharedVertices

File: Virtual Keyboard/Assets/ArcMeshDrawer.cs [hunk 2]
```diff
@@ -5,172 +5,156 @@ public class ArcMeshDrawer : MonoBehaviour {
...
-	public void calculateMesh () {
-		calculateSizes ();
-		float 	r = 3f,
-		x = 1f,
-		y = 0f,
-		theta = 0f;
+	private Vector3[] createArc( float radius ) {
+		Vector3[] arcPoints = new Vector3[sharedVertices.Length];
+		float 	x = 1f,
+				y = 0f,
```
Comment (eh-dub) [inline Virtual Keyboard/Assets/ArcMeshDrawer.cs:89]: Should the spacing be fixed here?

File: Virtual Keyboard/Assets/ArcMeshDrawer.cs [hunk 3]
```diff
@@ -5,172 +5,154 @@ public class ArcMeshDrawer : MonoBehaviour {
...
-		x = r*Mathf.Cos(arcLength);
-		y = r*Mathf.Sin(arcLength);
+		x = radius*Mathf.Cos(arcLength);
+		y = radius*Mathf.Sin(arcLength);
 		pos = new Vector3(x, y, 0);
-		vertices [size-1] = pos;
+		arcPoints [sharedVertices.Length - 1] = pos;
+		return arcPoints;
+	}
+	private void computeSharedVerticies() {
```
Comment (eh-dub) [inline Virtual Keyboard/Assets/ArcMeshDrawer.cs]: Should be computeSharedVertices()

File: Virtual Keyboard/Assets/ArcMeshDrawer.cs (hunk 2 above)
Comment (mroukema) [inline Virtual Keyboard/Assets/ArcMeshDrawer.cs:89]: Can't fix this this is just an issue with rendering on git hub

**Pull request merged**